docs/archive-redundancy.md

```diff
@@ -0,0 +1,63 @@
+
+    `ALTER DATABASE <DATABASE NAME> SET DEFAULT_TRANSACTION_ISOLATION TO SERIALIZABLE ;`
+
+This should be done after creating the database as described in [https://minaprotocol.com/docs/archive-node](https://minaprotocol.com/docs/archive-node) and before connecting an archive process to it.
+
+## Backing up block data
+
+Despite running redundant daemons and archive process, it may happen so that some blocks are still missed in which case you may want to make use of the following features to save and restore block data
+
+1. Upload block data to google cloud storage: To indicate a daemon to upload block data to google cloud storage, pass the flag `--upload-blocks-to-gcloud` . To successfully upload the file, daemon requires the following environment variables to be set:
+    1. `GCLOUD_KEYFILE` : Key file for authentication
+    2. `NETWORK_NAME`: Network name to be used in the filename
+    3. `GCLOUD_BLOCK_UPLOAD_BUCKET` : Google cloud storage bucket where the files are uploaded
+
+    The daemon generates a file for each block with the name `<network-name>-<protocol-state-hash>.json` . These are called precomputed blocks and will have all the fields of a block.
+
+2. Save block data from logs: The daemon also logs block data if the flag `-log-precomputed-blocks` is passed. The log to look for is `Saw block with state hash $state_hash` that contains `precomputed_block` in the metadata and has the block information. This is the same information (precomputed blocks) that gets uploaded to google cloud storage.
+3. Generate block data from another archive database: From a fully synced archive database, one can generate block data for each block using the `missing_subchain` tool. 
+
+    The tool takes an `--archive-uri`, a `--end-state-hash`, and an optional --start-state-hash and writes all the blocks in the chain starting from start-state-hash and ending at end-state-hash (including start and end).
+    If you provide both hashes, you get all blocks between start and end, and including the start and end. If just the end hash is provided, then it generates blocks starting with the unparented block closest to the end block, possibly the genesis block. The tool generates a file with name `<protocol-state-hash>.json` for each block. The block data in these files are called extensional blocks. Since these are generated from the database, they would have only the data stored in the archive database and would not contain any other information pertaining to a block (for example, blockchain snark)
+
+    Currently this tool is not packaged and can only be build from source using `make missing_subchain` 
+
+## Identifying missing blocks
+
+The tool `missing_block_auditor.exe` can be used to determine any missing blocks in an archive database. The tool outputs a list of state hashes of all the blocks that are missing. This can be used to monitor the archive database for any missing blocks.
```

Comment:
To be precise, the tool identifies missing parent blocks. There can be other missing blocks in subchains that end in the missing parents.